sklearn/decomposition/tests/test_kernel_pca.py

```diff
@@ -272,3 +273,19 @@ def test_nested_circles():
     # The data is perfectly linearly separable in that space
     train_score = Perceptron(max_iter=5).fit(X_kpca, y).score(X_kpca, y)
     assert_equal(train_score, 1.0)
+
+
+def test_kernel_conditioning():
+    """Test that ``check_psd_eigenvalues`` is correctly called."""
+
+    # create a pathological X leading to small non-zero eigenvalue
+    X = [[5, 1],
+         [5+1e-8, 1e-8],
+         [5+1e-8, 0]]
+    kpca = KernelPCA(kernel="linear", n_components=2,
+                     fit_inverse_transform=True)
+    kpca.fit(X)
+
+    # check that the small non-zero eigenvalue was correctly set to zero
+    assert kpca.lambdas_.min() == 0
+    assert_array_equal(kpca.lambdas_, check_psd_eigenvalues(kpca.lambdas_))
```

Comment:
Use `assert np.allclose()` instead

Reply:
why ? it should be exactly the same array (not close, exact)

Reply:
ok